docs/examples/DataSet/Data acquisition with help of threads .ipynb

```diff
@@ -0,0 +1,602 @@
+   "source": [
+    "In this notebook, we will explore how threading can be used with measurement context manager or dond functions for faster data acquisition. It is important to note that, QCoDeS provided threading happens per instrument. Meaning, per instrument one thread is created and all parameters from same instrument gets assigned to the same thread for data acquizition. \n",
+    "\n",
+    "Let us begin with some necessary imports."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "enormous-virus",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-24T15:42:55.906168Z",
+     "start_time": "2021-06-24T15:42:52.538980Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Logging hadn't been started.\n",
+      "Activating auto-logging. Current session state plus future input saved.\n",
+      "Filename       : C:\\Users\\v-singak\\.qcodes\\logs\\command_history.log\n",
+      "Mode           : append\n",
+      "Output logging : True\n",
+      "Raw input log  : False\n",
+      "Timestamping   : True\n",
+      "State          : active\n",
+      "Qcodes Logfile : C:\\Users\\v-singak\\.qcodes\\logs\\210624-8444-qcodes.log\n"
+     ]
+    }
+   ],
+   "source": [
+    "%matplotlib inline\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "import time\n",
+    "\n",
+    "import qcodes as qc\n",
+    "from qcodes import (\n",
+    "    Station,\n",
+    "    load_or_create_experiment,\n",
+    "    initialise_database,\n",
+    "    Measurement,\n",
+    "    Parameter,\n",
+    "    ParameterWithSetpoints\n",
+    ")\n",
+    "from qcodes.tests.instrument_mocks import DummyInstrument, DummyInstrumentWithMeasurement\n",
+    "from qcodes.dataset.plotting import plot_dataset\n",
+    "from qcodes.utils.validators import Numbers\n",
+    "from qcodes.utils.threading import call_params_threaded\n",
+    "from qcodes.utils.dataset.doNd import do1d"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fifty-metro",
+   "metadata": {},
+   "source": [
+    "Now, setup some instruments!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "divine-welsh",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-24T15:43:03.046647Z",
+     "start_time": "2021-06-24T15:43:03.033632Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "dac = DummyInstrument('dac', gates=['ch1', 'ch2'])\n",
+    "dmm1 = DummyInstrumentWithMeasurement(name='dmm1', setter_instr=dac)\n",
+    "dmm2 = DummyInstrumentWithMeasurement(name='dmm2', setter_instr=dac)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "dietary-coordinate",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-24T15:43:06.991369Z",
+     "start_time": "2021-06-24T15:43:06.971369Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "class SleepyDmmExponentialParameter(Parameter):\n",
```

Comment:
Perhaps we should explain that we on purpose make this instrument slow to simulate slow communication with the instrument